types/angular/index.d.ts

```diff
@@ -1029,6 +1029,7 @@ declare namespace angular {
         all<T1, T2, T3>(values: [T1 | IPromise<T1>, T2 | IPromise<T2>, T3 | IPromise<T3>]): IPromise<[T1, T2, T3]>;
         all<T1, T2>(values: [T1 | IPromise<T1>, T2 | IPromise<T2>]): IPromise<[T1, T2]>;
         all<TAll>(promises: Array<IPromise<TAll>>): IPromise<TAll[]>;
+        all<TAll>(values: TAll[]): IPromise<TAll[]>;
```

Comment:
Why not modify the previous line?
```ts
all<TAll>(promises: Array<T | IPromise<T>>): IPromise<T[]>;
```

Reply:
When I tried that, I got a bunch of cases where TypeScript was inferring the wrong type, assuming that the type was IPromise<IPromise<T>>.  By making it a separate line, it matches the IPromise<T> when that's what's wanted.

Reply:
I asked because that's how native promises are typed in TS. What's your compiler version and settings? Could you isolate such a case where you got that error?

Reply:
I looked into it more deeply, and I'd had issues with

```
all<TAll>(promises: Array<TAll>|Array<IPromise<TAll>>): IPromise<TAll[]>
```

not with what you suggested.  Updated the PR.